File: Python/Searching/LinearSearch.py
def LinearSearch(array , search):
    
    index = -1
    for i in range(0,len(array)):
        if(i==search):
            index==i

    if(i<0):
        print("Searched element is not available in the array")
    else:
        print("Searched element is available at index "+i)
        
arr = [2,3,4,10,40]
search = 10

LinearSearch(arr,search)

# Time Complexity -
# Best Case - O(1)
# Worst Case - O(n)
# Average Case - O(n/2)

# Space complexity - O(1)

File: JavaScript/Searching/LinearSearch.js
LinearSearch = (search,array) => {
    let index = -1;
    console.log(array);
    console.log(array.length);
    for(let i=0;i<array.length;i++)
        if(array[i]===search)
            index=i; 

    if(index<0)
        console.log("Searched element is not available in array");
    else
        console.log("Searched element is available in array at index "+index);
};

let array = [1,2,5,4,3];
let search = 5;

LinearSearch(search,array);

// Time Complexity -
// Best Case - O(1)
// Worst Case - O(n)
// Average Case - O(n/2)

// Space complexity - O(1)
